### PR TITLE
ref(mypy): Partial typing of MonitorEndpoint

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/monitors/endpoints/base.py [hunk 1]
```diff
@@ -44,15 +44,11 @@ class MonitorEndpoint(Endpoint):
...
-        organization_slug: str,
-        monitor_slug: str,
-        checkin_id: str | None = None,
-        *args,
-        **kwargs,
-    ):
+    def convert_args(self, request: Request, *args, **kwargs):
+        organization_slug: str = kwargs["organization_slug"]
+        monitor_slug: str = kwargs["monitor_slug"]
+        checkin_id: str | None = kwargs["checkin_id"]
```
Comment (wedamija) [inline src/sentry/monitors/endpoints/base.py:50]: I don't understand why we'd reduce the signature here, this is less descriptive

Comment (JoshFerge) [reply src/sentry/monitors/endpoints/base.py:50]: you can't override the signature of a subclass https://stackoverflow.com/questions/54346721/mypy-argument-of-method-incompatible-with-supertype

Comment (asottile-sentry) [reply src/sentry/monitors/endpoints/base.py:50]: there's a brand new thing for this: https://typing-extensions.readthedocs.io/en/latest/#override

Comment (untitaker) [reply src/sentry/monitors/endpoints/base.py:50]: I don't think override helps with this usecase. We have multiple usecases where we _need_ to violate the liskov substitution principle, and this is one of them. Another one is `BaseView.dispatch`

**Pull request closed without merging**